src/workers/AGENTS.md

```diff
@@ -0,0 +1,380 @@
+# Background Workers Documentation
+
+## Overview
+
+Our architecture follows a **reactive pattern** where we offload non-critical, asynchronous work to background workers. This approach improves API response times, increases system resilience, and enables better scalability.
+
+### Architecture Components
+
+- **Google Pub/Sub**: Message queue for asynchronous event processing
+- **Pulumi**: Infrastructure as Code (IaC) for managing subscriptions (located in `.infra/`)
+- **Topics**: Managed in a separate infrastructure repository
+- **Debezium + CDC**: Change Data Capture for reacting to database changes
+- **TypeScript Workers**: Type-safe worker implementations
+
+## When to Use Workers
+
+Use background workers when:
+
+1. **Non-mission-critical processing**: Work that doesn't need to complete before returning a response to the user
+2. **Async operations**: Tasks that can happen independently of the main request flow
+3. **Third-party integrations**: External API calls (Slack, SendGrid, etc.) that shouldn't block user requests
+4. **Heavy computations**: Image processing, data transformations, analytics calculations
+5. **Distributed transactions**: When you need to coordinate changes across multiple systems without blocking the primary transaction
+
+### Examples of Worker Use Cases
+
+- Sending notifications (email, push, real-time)
+- Updating reputation scores
+- Processing images and media
+- Syncing data to external services (CIO, Slack)
+- Analytics and metrics collection
+- Cleanup operations
+- Content processing (markdown, translations)
+
+## Worker Types
+
+### Typed Workers (`TypedWorker`)
```

Comment:
Wondering if we should also mention itr on the main one, that we prefer strong typed (can't recall if i saw it)